### PR TITLE
Make yowie playable

### PR DESCRIPTION
**Pull request by AW-FulCode**
## About the PR

I changed the Yowie to have some better bonuses to make up for their severe limitations and add RP value. I also adjusted one of those limitations.

My changes include:
- Making the heat resistance equal to that of Vulps (they have fur too)
- Slightly nerfing their cold resistance
- Making them immune to non-metabolised poison attacks (of which there are not all that many)
- Giving them their own organs, organ prototypes, and metabolisms
- Removing **poison** damage on all metabolism effects, making them immune to certain reagent effects (such as ForcedSleep), removing most medicine overdose damage/effects, and giving them comprehensive healing from most narcotics
- Slightly increased their unarmed attack (to 10) and gave them a wideswing alt attack
- Added a component for bypassing wielding requirements to shoot/shoot accurately and gave it to them

## Why / Balance

Yowie is a fun RP race to play, and I am very glad they got added. However there a few things with them that I had found lacking both in terms of roleplaying a Yowie as I'd expect them to be and viability in most combat scenarios.

The Yowie currently has the following limitations from a balance perspective:
- No external wear, including hardsuits and almost all armour. This is by far their greatest weakness, which I do not believe had been adequetly balanced for.
- No shoes. Arguably their second greatest weakness. No slip resistance, no grav loss measures, no foot protection. Diona at least are slow most of the time to not need slip resistance and don't need to worry about foot protection (correct me if I am wrong). And Diona can wear hardsuits and come back from the dead.
- Massive heat weakness, equal to moths and second only to Diona (who are made of wood). Fun fact, fur is insulative. And Yowies are Australian. It's hot over here man.
- They are big. I haven't done any kind of hitbox analysis tbh, I just assume they are a bigger target lmk if that's not true.

Strengths:
- Their unarmed attack is a crowbar that can't wideswing. It's not great.
- They have some good physical resistances, these are all pretty decent. Does not really make up for the inability to wear armour IMO but still good as they are.
- Except the poison resistance, which would be good if it did anything. There might be some attacks that deal poison damage without metabolisation, that's it. If you spike their drink, they're going down like any human. I believe this may have been a mistake.
- Cold resistance is alright.
- Rad resistance is nice to counteract complete lack of rad suits.

I, and I think others, don't really think these strengths make up for the lack of external wear and shoes. 

Each change I have made has been done for the follwing reasons:

**Heat mod change**: Vulps have fur, their heat mod is 1.15. I think that's enough for Yowie's too. I slightly nerfed the cold resistance to make up for a better heat mod, up from 0.7 to 0.85.

**Poison immunity and narcotics healing**: Making them immune to practically all poison was partly for RP and partly as something to make them unique. Unlike skeletons and silicons, they still metabolise toxins. And they still take rads, caustic, and cellular. But giving them essential immunity to poison damage specifically, most non-damage negative effects, and removing OD limits for them allow them to be the walking drug oceans they were always supposed to be. 

The character's they are based on will drink and eat old petrol and cleaning fluids among other dangerous substances. This is fun. When I first played Yowie, I wanted to get right in to the role and drink welding fuel. But doing so just slowly killed me all the same. 

From a balance perspective, it makes them slightly more viable as antags and quite a bit more viable as sec/command. A nocturine attack on a Yowie will not work, a strength somewhat making up for the vulnerability of not being able to wear a captain's carapace, for example. The downside is they cannot be put to sleep for surgery (not sure if that has any consequences here yet, probably should).

Narcotics healing is not too powerful, aside from healing radiation. They get the same healing from alcohol as dwarves, tricord like healing (with rads added) from most narcotics, and a heavily reduced form of the former healing with nicotine. Its useful enough to give them a slight boost, and fits in with the role. The rad healing is more to supplement the lack of radsuits.

**Melee increase and wideswing**: when I, as a secoff, see a Yowie entering the scene my first thought is, and should be, "uh oh, better prepare for the big guy here". But then it immediately turns to "oh wait, no they suck lmao". Which is really sad. A yowie should make a good bouncer/bodyguard, and you shouldn't want to get in to a fight with them. That's all really, it is pretty powerful from the perspective of unarmed combat, though should not be particularly useful for most antag/sec encounters.

**One-hand wielding**: quite powerful for sec and nukies, I added this pretty much so they would have some way of viability as nukies. Sure they can't wear hardsuits, so the challenge of getting to the station and dealing with spacing is still present. But raid ops exist and barozine has no negative effects on Yowies (including no screaming). 

They can now wield a shield (or any off hand item) and a gun without accuracy penalties. This gives them plenty of options to deal with their lack of armour. Combine that with meth and they could potentially be quite formidable. They may force a nukie team in to a specific strategy, but if their bonuses stack up enough I think they might be viable (nothing has been changed RE the roll for nukies as Yowie at this stage).

In summary: Yowies are now what they were always meant to be - big toxin sponges with yeti strength. Their combat bonuses make up for the massive penalties of no armour and no shoes. Their poison immunity and drug synergy make them more interesting and lore-friendly. I don't think this will make them too powerful, though I am open to feedback. But honestly I think is a much needed boost. And funny.

## Technical details

Metabolisms of all reagent prototypes that have negative effects have been changed to have Yowie specific conditions/exclusions.

The Yowie now has its own set of organs, body prototype, and metabolism type.

There is a NoWieldNeeded component with an optional GetBonus parameter (default true). The involved systems have been modified to apply gun bonuses on weapon pick up, as well as checks when shooting if the weapon has wielding requirements.

The Yowie prototype has been changed to make use of the new component and the new body prototype.

Guide entry has been adjusted.

**Changelog**

:cl:
- tweak: Yowies hit a bit harder and can wideswing!
- tweak: Yowies don't take as much heat damage anymore, but are also not as cold resistant.
- tweak: Yowies are immune to most forms of poison and can heal from most narcotics!
- add: Yowies can now use ranged weapons in one hand without any penalties!

### CODE REVIEW

File: Content.Shared/Weapons/Ranged/Components/NoWieldNeeded.cs
Comment (Aidenkrz) [inline Content.Shared/Weapons/Ranged/Components/NoWieldNeeded.cs:1]: Should be in Goob's namespace

File: Content.Shared/Weapons/Ranged/Events/GunRefreshModifiersEvent.cs [hunk 1]
```diff
@@ -19,5 +19,6 @@ public record struct GunRefreshModifiersEvent(
     Angle MinAngle,
     int ShotsPerBurst,
     float FireRate,
-    float ProjectileSpeed
+    float ProjectileSpeed,
+    EntityUid? User
```
Comment (Aidenkrz) [inline Content.Shared/Weapons/Ranged/Events/GunRefreshModifiersEvent.cs:23]: add a goob comment

File: Content.Shared/Weapons/Ranged/Systems/SharedGunSystem.cs [hunk 2]
```diff
@@ -549,7 +549,7 @@ public void CauseImpulse(EntityCoordinates fromCoordinates, EntityCoordinates to
         Physics.ApplyLinearImpulse(user, -impulseVector, body: userPhysics);
     }
 
-    public void RefreshModifiers(Entity<GunComponent?> gun)
+    public void RefreshModifiers(Entity<GunComponent?> gun, EntityUid? User = null)
```
Comment (Aidenkrz) [inline Content.Shared/Weapons/Ranged/Systems/SharedGunSystem.cs:552]: goob comment

File: Content.Shared/Weapons/Ranged/Systems/SharedGunSystem.cs [hunk 3]
```diff
@@ -565,7 +565,8 @@ public void RefreshModifiers(Entity<GunComponent?> gun)
             comp.MinAngle,
             comp.ShotsPerBurst,
             comp.FireRate,
-            comp.ProjectileSpeed
+            comp.ProjectileSpeed,
```
Comment (Aidenkrz) [inline Content.Shared/Weapons/Ranged/Systems/SharedGunSystem.cs:568]: goob comment

File: Content.Shared/Wieldable/WieldableSystem.cs [hunk 4]
```diff
@@ -42,7 +42,7 @@ public override void Initialize()
 
         SubscribeLocalEvent<WieldableComponent, UseInHandEvent>(OnUseInHand, before: [typeof(SharedGunSystem)]);
         SubscribeLocalEvent<WieldableComponent, ItemUnwieldedEvent>(OnItemUnwielded);
-        SubscribeLocalEvent<WieldableComponent, GotUnequippedHandEvent>(OnItemLeaveHand);
+        SubscribeLocalEvent<WieldableComponent, GotUnequippedHandEvent>(OnItemLeaveHand);
```
Comment (Aidenkrz) [inline Content.Shared/Wieldable/WieldableSystem.cs:45]: goob comment

File: Content.Shared/Wieldable/WieldableSystem.cs [hunk 5]
```diff
@@ -54,6 +54,7 @@ public override void Initialize()
         SubscribeLocalEvent<GunWieldBonusComponent, ItemUnwieldedEvent>(OnGunUnwielded);
         SubscribeLocalEvent<GunWieldBonusComponent, GunRefreshModifiersEvent>(OnGunRefreshModifiers);
         SubscribeLocalEvent<GunWieldBonusComponent, ExaminedEvent>(OnExamine);
+        SubscribeLocalEvent<GunWieldBonusComponent, GotEquippedHandEvent>(OnItemInHand);
```
Comment (Aidenkrz) [inline Content.Shared/Wieldable/WieldableSystem.cs:57]: goob comment

File: Content.Shared/Wieldable/WieldableSystem.cs [hunk 6]
```diff
@@ -70,8 +71,14 @@ private void OnMeleeAttempt(EntityUid uid, MeleeRequiresWieldComponent component
 
     private void OnShootAttempt(EntityUid uid, GunRequiresWieldComponent component, ref ShotAttemptedEvent args)
     {
+        if (TryComp<NoWieldNeededComponent>(args.User, out var noWieldNeeded) && noWieldNeeded.GetBonus) {
```
Comment (Aidenkrz) [inline Content.Shared/Wieldable/WieldableSystem.cs:74]: you get the idea, put a comment on all the changes that arent in the goob namespace

File: Resources/Locale/en-US/metabolism/metabolizer-types.ftl [hunk 7]
```diff
@@ -7,5 +7,6 @@ metabolizer-type-vox = Vox
 metabolizer-type-rat = Rat
 metabolizer-type-plant = Plant
 metabolizer-type-dwarf = Dwarf
+metabolizer-type-yowie = Yowie
```
Comment (Aidenkrz) [inline Resources/Locale/en-US/metabolism/metabolizer-types.ftl:10]: this should be in the goob namespace

File: Resources/Prototypes/Chemistry/metabolizer_types.yml [hunk 8]
```diff
@@ -44,3 +44,7 @@
 - type: metabolizerType
   id: Arachnid
   name: metabolizer-type-arachnid
+
+- type: metabolizerType
```
Comment (Aidenkrz) [inline Resources/Prototypes/Chemistry/metabolizer_types.yml:48]: should also be in goob namespace

**Review by Aidenkrz — CHANGES_REQUESTED** [review]

**Comment by WhiteGorila** [comment]
MERGE

File: Content.Shared/Wieldable/WieldableSystem.cs (hunk 4 above)
Comment (AW-FulCode) [reply Content.Shared/Wieldable/WieldableSystem.cs:45]: Looks like some whitespace got added here, I'll just try to cut that out

**Comment by Aidenkrz** [comment]
God this is just begging for merge conflicts but whatever.

**Pull request merged**